lib/proxy_api/resource.rb

```diff
@@ -84,5 +84,11 @@ def put(payload, path = "")
     def delete(path)
       resource[path].delete
     end
+
+    def cached_get(path = nil, payload = {})
+      Rails.cache.fetch("proxies/#{url}", :expires_in => 3.minutes) do
+        get(path, payload)
```

Comment:
I would assume we should cache on the controller level.. does clicking of refresh features invalidate the cache?